osu.Game.Tests/Visual/TestCaseBeatmapInfoWedge.cs

```diff
@@ -126,14 +127,31 @@ private void testNullBeatmap()
             AddAssert("check no infolabels", () => !infoWedge.Info.InfoLabelContainer.Children.Any());
         }
 
+        private void testAsyncLoading()
+        {
+            RulesetInfo firstRuleset = new OsuRuleset().RulesetInfo;
+            RulesetInfo secondRuleset = new TaikoRuleset().RulesetInfo;
+
+            var firstBeatmap = createTestBeatmap(firstRuleset, 25000000);
+            var secondBeatmap = createTestBeatmap(secondRuleset, 2000);
+
+            AddStep("select two beatmaps", () =>
+            {
+                infoWedge.Beatmap = beatmap.Value = new TestWorkingBeatmap(firstBeatmap);
+                infoWedge.Beatmap = beatmap.Value = new TestWorkingBeatmap(secondBeatmap);
+            });
+            AddUntilStep(() => infoWedge.Info.Working.Beatmap == secondBeatmap, "wait for load");
```

Comment:
This needs to wait for both beatmaps to load. Otherwise you aren't really checking if the beatmap that takes longer to load but is no longer current wouldn't still overwrite the info. That's why the logic to check for loaded beatmaps count was there.

Reply:
Right now it read ``Wait for second beatmap to load``, ``Assert second beatmap is loaded`` which wouldn't ever fail

Reply:
It was pointless and overly complex testing. And you were relying on arbitrary adding-more-objects making things slower. 

I'm PRing this as my own PR without this stuff.